### PR TITLE
Update contribution guidelines for libphonenumber-supporting-docs

### PR DESCRIPTION
**Pull request by keghani**
This change is [Reviewable](https://reviewable.io/reviews/googlei18n/libphonenumber/1624)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -46,7 +46,8 @@ If the evidence is publicly available online, please provide the link.
 
 If the evidence is not publicly available online, make sure that you have the
 rights to share this with us, and confirm this by reading and signing the
-appropriate Contributor License Agreement (CLA):
+appropriate Contributor License Agreement (CLA); **use an email address
```
Comment (lararennie) [inline CONTRIBUTING.md]: Can you use a new sentence for this? it gets a bit hard to read. I'd do it after the bullet points as "N.B. Use an email address associated with the GitHub username you used to filed the corresponding issue" (but still in bold).

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -58,13 +59,16 @@ information with us, and that we may use, modify, reproduce, publicly display,
...
+Once you have signed the CLA:
+
+*   If there is no open validation issue for your request, create one.
+*   Comment on the open issue, e.g. "I signed the CLA, please whitelist me to
+    send private documentation to libphonenumber-supporting-docs".
+*   Someone from the team will whitelist you and update you on the issue.
+
+Once whitelisted, you will be able to email
+libphonenumber-supporting-docs@googlegroups.com. Only the core libphonenumber
+team has access to view emails sent to this address.
```
Comment (lararennie) [inline CONTRIBUTING.md]: Should we mention something about who the core libphonenumber team is, i.e. that it is restricted to Google employees?

Comment (keghani) [inline CONTRIBUTING.md]: I could link to https://github.com/orgs/googlei18n/teams/libphonenumber but we don't update the 2 lists concurrently so adding it here would just be a statement we couldn't uphold. E.g. not everyone in that list reads the supporting docs.

File: CONTRIBUTING.md (hunk 1 above)
Comment (keghani) [inline CONTRIBUTING.md]: Done

**Pull request merged**